core/dive.c

```diff
@@ -270,8 +270,7 @@ enum divemode_t get_current_divemode(struct divecomputer *dc, int time, struct e
 	return *divemode;
 }
 
-/* this returns a pointer to static variable - so use it right away after calling */
```

Comment:
this comment is still true... why did you delete it?

Reply:
Deleting it makes sense - if it's const and nobody will change it, it doesn't matter that it's static.

Reply:
Also note that the comment was never correct in the general case - only if event is not a gas-change event would it return pointer-to-static. The other case (pointer-to-heap) seems much more dangerous and is probably the one that deserves a comment.

Even though it's not classical C, I wonder if we could simply return by value here. `gasmix` is two integers. I think on common 64-bit platforms this will be returned in register anyway?

For
```
struct X {
        int a;
        int b;
};

struct X f()
{
        struct X x;
        x.a = 1;
        x.b = 2;
        return x;
}
```
I get
```
f:
        movabsq $8589934593, %rax
        ret
```